### ISSUE TITLE
[consensus] Add round timeout debugging information

### ISSUE THREAD

**Issue by aching** [libra/libra#1702]
When a round timeout occurs, one has to parse the logs to determine what steps were completed and understand why the timeout occurred.  This is time consuming and not all the logs are available in different debugging modes.  Given that round timeouts occur at most every second, it is reasonable to print out what steps in the round were completed to help diagnose the issue.  For example, at a high level, here are the steps below:

if leader:
(synced to round proof, received_all_votes, proposed, executed_proposal, voted)
else:
(synced to round proof, received_proposal, synced_dependencies, executed_proposal, proposed, voted)

We could have the timeout period, step we are waiting for to help with debugging why the timeouts are occurring.

Basically we can augment the current code:

        warn!(
            "Round {} timed out: {}, expected round proposer was {:?}, broadcasting the vote to all replicas",
            round,
            if last_vote_round == round { "already executed and voted at this round" } else { "will try to generate a backup vote" },
            self.proposer_election.get_valid_proposers(round).iter().map(|p| p.short_str()).collect::<Vec<String>>(),
        );
